tests/console/commands/self/test_add_plugins.py

```diff
@@ -194,7 +194,7 @@ def test_add_existing_plugin_warns_about_no_operation(
     installed: TestRepository,
 ) -> None:
     pyproject = SelfCommand.get_default_system_pyproject_file()
-    with open(pyproject, "w", encoding="utf-8", newline="") as f:
+    with pyproject.open("w", encoding="utf-8", newline="") as f:
```

Comment:
Can we use write_text? Similar in a few other places.

Reply:
Only after dropping support for Python 3.9, I assume:

> Changed in version 3.10: The newline parameter was added.

https://docs.python.org/3/library/pathlib.html#pathlib.Path.write_text

Reply:
Exactly this. I can add a TODO for when we drop 3.9 to change that